### ISSUE TITLE
Invalid GET request causes 500

### ISSUE THREAD

**Issue by JRaspass** [PerlDancer/Dancer2#1496]
```
$ plackup -e 'use Dancer2; dance'
HTTP::Server::PSGI: Accepting connections at http://0:5000/
127.0.0.1 - - [14/Feb/2019:14:29:25 +0000] "GET / HTTP/1.1" 500 180 "-" "curl/7.64.0"
```
```
$ curl -so /dev/null -w "%{http_code}\n" -H 'Content-Type: multipart/form-data; boundary=a' -k http://localhost:5000
500
```

It seems if you pass a GET request which obviously has no body, but claim to have multipart/form-data in your body then Dancer2 dies trying to read said non-existent body.

This is similar in spirit to #585 and I think we should either skip processing the body when we don't have one, or return a 400.

**Comment by SysPete** [comment]
400 response seems sane in this case.

**Comment by cromedome** [comment]
This is resolved by #1509. Closing!
